### DISCUSSION TITLE
Separate out mixins from components - enhancement

### DISCUSSION THREAD

**Discussion by nbannist** [indiana-university/rivet-source#428] [Feature and enhancement requests]
## Code Enhancement

Using: Rivet 1.2.0

As I've been developing styles for the new and future Knowledge Base frontend, which will be a Rivet-based app, I've noticed that some Mixins are inside component SCSS files. This makes it difficult to extend project styles with Rivet styles to keep things consistent without also unnecessarily including Rivet style code.

There are also comments in the Rivet code base about making theses Mixins available so outside projects can more easily extend their styles using Rivet Mixins to make code Drier and less likely to break when Rivet is updated. I haven't found a ticket about this yet though.

## Additional Info

- Links to research, documentation, or code examples
  * [Lists Mixins within component code (lines 31 and 36)](https://github.com/indiana-university/rivet-source/blob/develop/src/sass/components/_lists.scss)
  * [Grid (comments on line 38 and 84)](https://github.com/indiana-university/rivet-source/blob/develop/src/sass/components/_grid.scss)

## Notes for discussion

 - What would the preferred structure of these Mixin files take? 
 - Should they have their own folder separated into components? 
 - Should they be separate files grouped with associated components? 
 - All mixins in one file?

**Comment by basham** [comment]
This discussion is somewhat related to what I started with #38. How can Rivet provide hooks into itself, so apps don't have to duplicate underlying things like variables or mixins?

**Comment by levimcg** [comment]
@nbannist That's a good point. I had originally intended to create more mixins that could also be used by developers using Rivet's Sass. Unfortunately, I just never got around to it. I would like to revisit though.

We haven't historically done a good job of documenting Sass usage, but we're currently working on adding documentation for the Sass variables (e.g. [this PR](https://github.com/indiana-university/rivet-docs/pull/15)) on the website for things like typography and spacing. We should have those docs live on the site soon.

**Comment by nbannist** [comment]
@levimcg The variables are mostly broken out (except for the grid stuff). It would be nice if the Mixins were as easily usable/includable too. Would be helpful for my team and I'm sure others'.

@basham Dang. How did I miss that ticket? :/
